google-cloud-storage/src/main/java/com/google/cloud/storage/Channel.java

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.storage;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * A Google cloud storage notification channel object used to watch the changes on objects in the
+ * {@code Bucket}.
+ */
+public class Channel implements Serializable {
```

Comment:
`Channel` is very generic name. Would you suggest a more specific class name, like BucketNotificationChannel
Channel implies read/write methods, but I don't see none of it, it looks like a collection of properties.
An example of usage would be very helpful here.

Reply:
I think ` Channel ` would be better here as `watchAll()` mthod returns instance of Channel.

Reply:
Also important, this is not a java.nio.channels.Channel and will be confused with that. NotificationChannel might work, but definitely not "Channel".